packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -95,7 +96,22 @@ class RootState {
       alwaysRevalidate: false,
     });
 
-    this.render = () => {
+    const handleError = (fn: () => void) => {
```

Comment:
I think this can be made just a bit better:

```js
let handleError;

if (DEBUG) {
  handleError = fn => {
    return () => {
      try {
        fn();
      } catch (e) {
        e.message += '\n  in -> ' + env.debugStack.toString();
        env.debugStack._stack = [];
        throw e;
      }
    };
  }
}

let render = () => {
  let layout = template.asLayout();
  let handle = layout.compile();
  let iterator = renderMain(
  let result = (this.result = iteratorResult.value);

  // override .render function after initial render
  this.render = DEBUG ? 
    handleError(() => result.rerender(options)) : 
    () => result.rerender(options);
};

this.render = DEBUG ? handleError(render) : render;
```

This way we don't include the extra function call, and the `handleError` function will be fully removed from production builds.